app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt

```diff
@@ -3980,4 +3980,78 @@ class PatientRepositoryAndroidTest {
         patientWithAnotherFacilityAsAssignedFacility, patientWithCurrentFacilityAsAssignedFacility
     )
   }
+
+  @Test
+  fun querying_all_patients_in_facility_must_return_list_of_patients_partitioned_by_assigned_facility_and_order_by_name() {
+    fun createPatientWithNameAndAssignedFacilityID(
+        patientUuid: UUID,
+        patientName: String,
+        assignedFacilityId: UUID?
+    ) {
+      val patientProfile = testData
+          .patientProfile(patientUuid = patientUuid, patientName = patientName, patientAssignedFacilityId = assignedFacilityId)
+          .let { patientProfile ->
+            patientProfile.copy(
+                patient = patientProfile.patient.copy(fullName = patientName, assignedFacilityId = assignedFacilityId),
+            )
+          }
+
+      patientRepository.save(listOf(patientProfile)).blockingAwait()
+    }
+
+    fun searchResults(facility: Facility): List<String> {
+      return patientRepository
+          .allPatientsInFacility2(facility)
```

Comment:
Can we think of a different name than `allPatientsInFacility2`?

Reply:
My plan is to eventually remove the original `allPatientsInFacility` and rename `allPatientsInFacility2` to `allPatientsInFacility`. But I can rename these to `allPatientsInFacility_Old` & `allPatientsInFacility` if this is a problem. WDYT?

Reply:
Ah, if we are planning to remove it later, then it doesn't really matter.